### PR TITLE
[RDY] Prison mapgen JSONify

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
Conversion of `prison` mapgen from `map::draw_map` to JSON.

### CODE REVIEW

File: data/json/itemgroups/locations.json [hunk 1]
```diff
@@ -83,6 +83,13 @@
       ["material_niter", 20],
       ["material_limestone", 5]
     ]
+  },{
+    "type":"item_group",
+    "id":"prison_textile",
+    "subtype": "collection",
+    "entries" : [
+      { "item": "blanket", "prob": 25 },
+      { "item": "jumpsuit", "prob": 35 }
```
Comment (Reclusive-reptile) [inline data/json/itemgroups/locations.json]: Should jumpsuit be replaced with striped_shirt and striped_pants? Prison wardens generally don't like inmates concealing stuff in ample pocket space.

Comment (ZhilkinSerg) [inline data/json/itemgroups/locations.json]: Jumpsuits were in original mapgen - see https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/mapgen.cpp#L7354.

I believe we better change loot once all mapgen is moved to json from cpp.

Comment (kevingranade) [inline data/json/itemgroups/locations.json]: I agree, open a PR or issue to track updating the prision definition to use new items if you want (I don't think those existed when the prison layout was added).

**Comment by Coolthulhu** [comment]
"Visions of solitude" spawned in wrong place - above the ground
The entire lower level has benches on solid rock (visible with a clairvoyance artifact).

Other than that looks good.

**Comment by codemime** [comment]
This has multiple merge conflicts (with your other PRs that have been already merged).

**Comment by ZhilkinSerg** [comment]
I believe it will be better to jsonize mapgen one at a time. I will create new pull request. Thanks, @codemime.

**Comment by Coolthulhu** [comment]
It's easy to mass-test them, the only problem is that they conflict with each other.

You could have dependent PRs: create one branch with one jsonized mapgen, then create new one on this branch rather than on master. This ensures no conflicts between those two branches.

**Comment by ZhilkinSerg** [comment]
Thanks, @Coolthulhu. I thought about this, but had some doubts. Branches should be created and merged in certain order, right? Also won't there be conflicts, if I add some changes in parent branch, but won't add these changes to child branches.

**Comment by Zireael07** [comment]
@ZhilkinSerg You can easily merge parent branch into child branch, though.

**Comment by ZhilkinSerg** [comment]
Superseded by #20408

**Pull request closed without merging**